### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by mati865**
During housekeeping I noticed few things:

- `!=` and `==` for floats provides the same suggestion:
```
warning: strict comparison of f32 or f64
   --> src/consts.rs:381:45
    |
381 |                 BiEq => Some(Constant::Bool(l == r)),
    |                                             ^^^^^^ help: consider comparing them within some error: `(l - r).abs() < error`
    |
warning: strict comparison of f32 or f64
   --> src/consts.rs:382:45
    |
382 |                 BiNe => Some(Constant::Bool(l != r)),
    |                                             ^^^^^^ help: consider comparing them within some error: `(l - r).abs() < error`
    |
```
I think `!=` should suggest `(l - r).abs() > error` so I changed it in the code that way. I'll fix it here or create next PR if somebody confirms that I'm correct.
- Remainder was broken for floats?
Replaced `BiRem => Some(Constant::F64(l * r))` with `BiRem => Some(Constant::F64(l % r))` similarly for `F32`.
I don't know if it's correct for Rust but basic examples work (with limited accuracy ofc).
- There are some warning left but I don't dare to touch it:

<details>
  <summary>Click to expand</summary>
 
```
warning: 5th binding whose name is just one char
   --> src/utils/hir_utils.rs:350:21
    |
350 |                 let c: fn(_, _, _) -> _ = ExprAssignOp;
    |                     ^
    |
    = note: #[warn(many_single_char_names)] on by default
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#many_single_char_names

warning: consider using `Option<T>` instead of `Option<Option<T>>` or a custom enum if you need to distinguish all 3 cases
   --> src/array_indexing.rs:105:13
    |
105 |     start: &Option<Option<Constant>>,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(option_option)] on by default
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#option_option

warning: consider using `Option<T>` instead of `Option<Option<T>>` or a custom enum if you need to distinguish all 3 cases
   --> src/array_indexing.rs:106:11
    |
106 |     end: &Option<Option<Constant>>,
    |           ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#option_option

warning: the function has a cyclomatic complexity of 31
   --> src/methods.rs:685:5
    |
685 | /     fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx hir::Expr) {
686 | |         if in_macro(expr.span) {
687 | |             return;
688 | |         }
...   |
771 | |         }
772 | |     }
    | |_____^
    |
    = note: #[warn(cyclomatic_complexity)] on by default
    = help: you could split it up into multiple smaller functions
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#cyclomatic_complexity

warning: this function has too many arguments (8/7)
   --> src/methods.rs:891:5
    |
891 | /     fn check_general_case(
892 | |         cx: &LateContext,
893 | |         name: &str,
894 | |         method_span: Span,
...   |
948 | |         );
949 | |     }
    | |_____^
    |
    = note: #[warn(too_many_arguments)] on by default
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#too_many_arguments
```
</details>

It still produces ICE unless all warnings are fixed/silenced.

### CODE REVIEW

**Comment by oli-obk** [comment]
retriggering travis and appveyor

**Comment by mati865** [comment]
Was about to ask for appveyor trigger but looks like it's going to timeout again for the [last commit in master branch](https://ci.appveyor.com/project/nrc/rust-clippy/build/1.0.497).

Before miri merge it would finish under 10 minutes. Maybe miri caused issues for Windows?

**Comment by oli-obk** [comment]
> Before miri merge it would finish under 10 minutes. Maybe miri caused issues for Windows?

I'll investigate

File: clippy_lints/src/inline_fn_without_body.rs [hunk 1]
```diff
@@ -37,12 +37,9 @@ impl LintPass for Pass {
...
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
     fn check_trait_item(&mut self, cx: &LateContext<'a, 'tcx>, item: &'tcx TraitItem) {
-        match item.node {
-            TraitItemKind::Method(_, TraitMethod::Required(_)) => {
-                check_attrs(cx, &item.name, &item.attrs);
-            },
-            _ => {},
-        }
+        if let TraitItemKind::Method(_, TraitMethod::Required(_)) = item.node {
+    check_attrs(cx, &item.name, &item.attrs);
```
Comment (oli-obk) [inline clippy_lints/src/inline_fn_without_body.rs]: formatting fail

File: clippy_lints/src/consts.rs [hunk 2]
```diff
@@ -377,9 +378,9 @@ impl<'c, 'cc> ConstEvalLateContext<'c, 'cc> {
                 BiSub => Some(Constant::F32(l - r)),
                 BiMul => Some(Constant::F32(l * r)),
                 BiDiv => Some(Constant::F32(l / r)),
-                BiRem => Some(Constant::F32(l * r)),
-                BiEq => Some(Constant::Bool(l == r)),
-                BiNe => Some(Constant::Bool(l != r)),
+                BiRem => Some(Constant::F32(l % r)),
+                BiEq => Some(Constant::Bool((l - r).abs() < f32::EPSILON)),
```
Comment (oli-obk) [inline clippy_lints/src/consts.rs]: these need to stay `==` and `!=`, just ignore the lint here

File: clippy_lints/src/ptr.rs [hunk 3]
```diff
@@ -225,20 +225,17 @@ fn check_fn(cx: &LateContext, decl: &FnDecl, fn_id: NodeId, opt_body_id: Option<
...
-                                    "using a reference to `Cow` is not recommended.",
-                                    |db| {
-                                        db.span_suggestion(arg.span, "change this to", "&".to_owned() + &r);
-                                    },
-                                );
-                            },
-                            None => (),
-                        }
+                        if let Some(r) = replacement {
+    span_lint_and_then(
```
Comment (oli-obk) [inline clippy_lints/src/ptr.rs]: formatting fail

File: clippy_lints/src/inline_fn_without_body.rs (hunk 1 above)
Comment (mati865) [reply clippy_lints/src/inline_fn_without_body.rs]: Interesting, it was created with rustfix from misformatted clippy suggestion.
I was sure rustfmt would fix formatting, I'm gonna be more careful with it now.

```
warning: you seem to be trying to use match for destructuring a single pattern. Consider using `if let`
  --> src/inline_fn_without_body.rs:40:9
   |
40 | /         match item.node {
41 | |             TraitItemKind::Method(_, TraitMethod::Required(_)) => {
42 | |                 check_attrs(cx, &item.name, &item.attrs);
43 | |             },
44 | |             _ => {},
45 | |         }
   | |_________^
   |
   = note: #[warn(single_match)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#single_match
help: try this
   |
40 |         if let TraitItemKind::Method(_, TraitMethod::Required(_)) = item.node {
41 |     check_attrs(cx, &item.name, &item.attrs);
42 | }
   |
```

File: clippy_lints/src/ptr.rs (hunk 3 above)
Comment (mati865) [reply clippy_lints/src/ptr.rs]: Like the upper one:
```
help: try this
    |
228 |                         if let Some(r) = replacement {
229 |     span_lint_and_then(
230 |         cx,
231 |         PTR_ARG,
232 |         arg.span,
233 |         "using a reference to `Cow` is not recommended.",
  ...
```

**Comment by mati865** [comment]
Should I squash those commits?

**Comment by mati865** [comment]
No matter what I do I cannot trigger dogfood anymore, hope it's what you meant by ignoring lint.

**Comment by oli-obk** [comment]
lgtm, thx

> No matter what I do I cannot trigger dogfood anymore, hope it's what you meant by ignoring lint.

Yea I really need to have a look at what's going on there.

**Pull request merged**